### PR TITLE
Shows which world your cannon is on

### PR DESCRIPTION
**Pull request by DanielOwen1**
cannonworld

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java [hunk 1]
```diff
@@ -200,6 +200,25 @@ default boolean showCannon()
...
+	default int cannonWorld()
+	{
+		return -1;
+	}
+
+	@ConfigItem(
+			keyName = "cannonWorld",
+			name = "",
+			description = ""
+	)
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java]: Too much indent above, should probably review the following article:
https://github.com/runelite/runelite/wiki/Checkstyle-with-IntelliJ-IDEA

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java [hunk 2]
```diff
@@ -200,6 +200,25 @@ default boolean showCannon()
 		return true;
 	}
 
+	// Stored Data
+	@ConfigItem(
+			keyName = "cannonWorld",
+			name = "",
+			description = "",
+			hidden = true
+	)
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java]: Too much indent above, should probably review the following article:
https://github.com/runelite/runelite/wiki/Checkstyle-with-IntelliJ-IDEA

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java [hunk 3]
```diff
@@ -200,6 +200,25 @@ default boolean showCannon()
 		return true;
 	}
 
+	// Stored Data
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java]: This comment doesn't contain any useful information so can be removed

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java [hunk 4]
```diff
@@ -516,7 +520,16 @@ public void createGameTimer(GameTimer timer)
 		removeGameTimer(timer);
 
 		TimerTimer t = new TimerTimer(timer, this);
-		t.setTooltip(timer.getDescription());
+
+		if (timer.getDescription().equals(("Cannon")))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: This is probably wrong way to do this, extend the existing API instead

Comment (DanielOwen1) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: Any way you would suggest changing this? I'm not really sure what to do differently

**Comment by DanielOwen1** [comment]
Added two functions. Not sure if that is what you meant.

**Comment by DanielOwen1** [comment]
@deathbeam Can you check the changes when you get the chance.

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java [hunk 5]
```diff
@@ -516,12 +517,22 @@ public void createGameTimer(GameTimer timer)
 		removeGameTimer(timer);
 
 		TimerTimer t = new TimerTimer(timer, this);
-		t.setTooltip(timer.getDescription());
+		t.setTooltip(getTooltipDescription(timer));
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: I did not meant moving the logic to separate function. I meant adding subclass of timer and make it like CannonTimer or something and sending the cannonWorld there.

**Comment by DanielOwen1** [comment]
@deathbeam I tried this first since it makes it more generic and removes the config stuff. I can always add the cannon class instead if you want

**Comment by deathbeam** [comment]
This looks fine yeah

**Comment by deathbeam** [comment]
Looks like this needs rebasing

**Comment by Nightfirecat** [comment]
This was added in #12530

**Pull request closed without merging**